Ufal dtq sync 2025 05 14

kosarko opened the pull request:
This contains:
- the acknowledgment of the funder
- ufal/dspace-angular#30 video preview part of ufal/clarin-dspace#1072


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced video preview with secure playback, including authorization checks and automatic token renewal for protected content.
  - Added an acknowledgment document with visual and textual credit for financial support.

- **Bug Fixes**
  - Improved image and video preview handling for different file formats.

- **Refactor**
  - Updated internal identifiers from numeric values to UUID strings across components and tests.
  - Adjusted file preview logic to fetch only original file types.

- **Tests**
  - Updated test setups to use UUID identifiers and improved dependency mocks for file preview components.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

kosarko committed:
Video files previews

This uses the thumbnail as poster (if available) and correctly sets the
source of the video

currently only works for anonymously accessible files.

(cherry picked from commit 4832c2f2661baa00326311aa39198a66b4bc811c)

kosarko committed:
Create Acknowledgment-ReadMe.md

Acnkowledgment of NRP project

(cherry picked from commit ad889b25549290c857b14abca73afc27a6463f8f)

kosarko committed:
Handle video previews for restricted items

append a shortlived token at the right time (error, seeking, stalled)

(cherry picked from commit 2c12d7d142a60a4be0a99b77df5cd4dc7d49d455)

kosarko committed:
Display only ORIGINAL bitstreams

Thumbnails, when available, should be shown istead of the generic
MIME_TYPE_IMAGE.

Content of the TEXT bundle should not be shown at all this is usually
automatically extracted "text layer" of a PDF, useful for indexing, but
don't want people downloading it.

(cherry picked from commit 3862442f003a2f568fbc652322d237c90ece1905)

kosarko committed:
fix linter and test errors

(cherry picked from commit f85209673074972c822645ca309f682cc43b6cf7)

milanmajchrak commented on src/app/item-page/simple/field-components/preview-section/file-description/file-description.component.html:
```diff
@@ -29,7 +31,8 @@
     </dl>
     <img
       *ngIf="fileInput.format !== 'video/mp4'"
-      [src]="MIME_TYPE_IMAGES_PATH + fileInput.format?.replace('/','-') + '.png'"
```
Following the CodeRabbit `There is an extra space before the opening parenthesis and after the closing one` `[src]=" ( thumbnail_url$ | async)`

milanmajchrak commented on src/app/item-page/simple/field-components/preview-section/file-description/file-description.component.spec.ts:
```diff
@@ -12,11 +12,20 @@ import { RouterTestingModule } from '@angular/router/testing';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { HALEndpointService } from '../../../../../core/shared/hal-endpoint.service';
 import { FileSizePipe } from '../../../../../shared/utils/file-size-pipe';
+import {BitstreamDataService} from '../../../../../core/data/bitstream-data.service';
```
not consistent import brackets

milanmajchrak commented on src/app/item-page/simple/field-components/preview-section/file-description/file-description.component.ts:
```diff
@@ -1,9 +1,20 @@
-import { Router } from '@angular/router';
-import { ConfigurationDataService } from '../../../../../core/data/configuration-data.service';
-import { getFirstSucceededRemoteData } from '../../../../../core/shared/operators';
+import {Component, ElementRef, Input, OnInit, ViewChild} from '@angular/core';
```
not consistent import brackets

milanmajchrak commented on src/app/item-page/simple/field-components/preview-section/file-description/file-description.component.ts:
```diff
@@ -30,6 +52,114 @@ export class FileDescriptionComponent implements OnInit {
+    if (video) {
+      const error$ = fromEvent(video, 'error');
+      error$.subscribe((event) => {
+        //console.log('error', video.error.message);
```
Do we need this commented out console.log code?

kosarko replied:
Is there something like log.debug, can I do the logging conditionally?

milanmajchrak replied:
I'm not sure how to do it. I think we don't have debug logging in this Angular project, except for this [config.yml](https://github.com/dataquest-dev/dspace-angular/blob/dtq-dev/config/config.yml#L1), which is used here: [app.metareducers.ts](https://github.com/dataquest-dev/dspace-angular/blob/dtq-dev/src/app/app.metareducers.ts#L6), as far as I know.

milanmajchrak reviewed (changes requested):
Please check the Code Rabbits review, potential issues.

kosarko committed:
Code review follow up

the listOfFiles should really not contain files from "TEXT" or
"THUMBNAIL" bundles.

kosarko committed:
code review

unsubscribe error$, seeking$ and $stalled

kosarko committed:
code review - thumbnail might be undefined

kosarko committed:
code review - consistent formatting
